packages/client/src/v2-events/features/workqueues/EventOverview/components/EventHistory/EventHistory.tsx

```diff
@@ -54,25 +54,94 @@ const messages = defineMessages({
   role: {
     id: 'v2.event.history.role',
     defaultMessage:
-      '{role, select, LOCAL_REGISTRAR{Local Registrar} other{Unknown}}',
+      '{role, select, LOCAL_REGISTRAR {Local Registrar} SOCIAL_WORKER {Field Agent} REGISTRATION_AGENT {Registration Agent} HEALTH {Health integration} IMPORT {Import integration} NATIONAL_ID {National ID integration} RECORD_SEARCH {Record search integration} WEBHOOK {Webhook} other {Unknown}}',
     description: 'Role of the user in the event history'
+  },
+  systemDefaultName: {
+    id: 'v2.event.history.systemDefaultName',
+    defaultMessage: 'System integration',
+    description: 'Fallback for system integration name in the event history'
   }
 })
 
+const Header = styled(Text)`
+  margin-bottom: 20px;
+`
+
+// At first we tried adding a field such as 'userType' to all actions etc, but that would have caused file changes to like 20 files.
+// So lets Keep It Stupid Simple and just check it there is no location -> machine user.
+function isUserAction(
+  item: ActionDocument
+): item is ActionDocument & { createdAtLocation: string } {
+  return Boolean(item.createdAtLocation)
+}
```

Comment:
👍 I had to also think this in the Postgres schema. I was considering doing

```sql
 created_by_system boolean GENERATED ALWAYS AS (created_at_location IS NULL),
 ```
 
 but when we bring systems and users to Postgres, we can join the details of the actor in a different way and this type of a column would be redundant. Lets remember to come back to this when users and systems are migrated to Postgres.